arbitrum-docs/for-devs/quickstart-solidity-hardhat.md

```diff
@@ -0,0 +1,370 @@
+| `scripts/`          | Contains scripts that you can use to interact with your smart contracts. You should see `deploy.js` here. |
+| `hardhat.config.js` | Contains the configuration settings for Hardhat.                                                          |
+
+
+Replace the contents of `hardhat.config.js` with the following[^6]:
+
+```javascript title="hardhat.config.js"
+require("@nomicfoundation/hardhat-toolbox");
+
+/** @type import('hardhat/config').HardhatUserConfig */
+module.exports = {
+  solidity: "0.8.18",
+  networks: {
+    hardhat: {
+      chainId: 1337
+    },
+    arbitrumGoerli: {
+      url: "https://goerli-rollup.arbitrum.io/rpc",
+      chainId: 421613
+      //accounts: [process.env.PRIVATE_KEY] TODOs
+    },
+    arbitrumOne: {
+      url: "https://arb1.arbitrum.io/rpc",
+      //accounts: [process.env.PRIVATE_KEY]
+    },
+    arbitrumNova: {
+      url: "https://arbnova.com/rpc",
+      //accounts: [process.env.PRIVATE_KEY]
+    }
+  }
+};
+```
+
+Run `npx hardhat compile` to compile the default `contracts`. You may be prompted to install additional dependencies - follow those instructions until this command runs successfully[^6]. You should see `Compiled 1 Solidity file successfully` in the terminal output. You should also see a new `decentralized-cupcakes/artifacts/` directory. This directory contains the compiled smart contract[^7].
+
+Open `scripts/deploy.js` and replace its contents with the following:
+
+```javascript title="scripts/deploy.js"
+const hre = require("hardhat");
+
+async function main() {
+
+  const VendingMachineFactory = await hre.ethers.getContractFactory("VendingMachine");
+  const vendingMachine = await VendingMachineFactory.deploy();
+  await vendingMachine.deployed();
+
+  console.log(
+    `Cupcake vending machine deployed to ${vendingMachine.address}`
+  );
+}
+
+main().catch((error) => {
+  console.error(error);
+  process.exitCode = 1;
+});
+```
+
+We'll use this to deploy our smart contract in a moment. Next, delete `contracts/Lock.sol` and replace it with `contracts/VendingMachine.sol`, the smarter alternative to our dumb Javascript contract:
+
+```solidity title="VendingMachine.sol"
+pragma solidity ^0.8.9;
+
+// Rule 2: The vending machine's rules can't be changed by anyone.
+contract VendingMachine {
+    // state variables = internal memory of the vending machine
+    mapping(address => uint) private _cupcakeBalances;
+    mapping(address => uint) private _cupcakeDistributionTimes;
+
+    function giveCupcakeTo(address userAddress) public returns (bool) {
+        if (_cupcakeBalances[userAddress] == 0) {
+            _cupcakeDistributionTimes[userAddress] = block.timestamp; // todo - decide whether or not there should be 1:1-ish mapping between js and solidity "contracts" for the sake of pattern recognition / learning; if so, mirror; if not, delete
```

Comment:
actually looking again: as-is it's a bug, not just a noop  — the initial attempt to claim will always fail, (block.timestamp + 5 seconds <= block.timestamp)